lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -391,12 +391,17 @@ class ModuleInterfaceLoaderImpl {
     // anyways.
     H = hash_combine(H, interfacePath);
 
-    // Include the target CPU architecture. In practice, .swiftinterface files
-    // will be in architecture-specific subdirectories and would have
-    // architecture-specific pieces #if'd out. However, it doesn't hurt to
+    // Include the normalized target triple. In practice, .swiftinterface files
+    // will be in target-specific subdirectories and would have
+    // target-specific pieces #if'd out. However, it doesn't hurt to
     // include it, and it guards against mistakenly reusing cached modules
-    // across architectures.
-    H = hash_combine(H, SubInvocation.getLangOptions().Target.getArchName());
+    // across targets. Note that we this normalization explicitly doesn't
+    // include the minimum deployment target (e.g. the '12.0' in 'ios12.0').
+    // Different clients with different minimum deployment targets can import
+    // the same .swiftmodule.
+    auto normalizedTarget =
+        getTargetSpecificModuleTriple(SubInvocation.getLangOptions().Target);
+    H = hash_combine(H, normalizedTargetTriple.str());
```

Comment:
Should `normalizedTargetTriple` be `normalizedTarget` instead?

Reply:
Oops, yes. Am in the process of rebuilding after the big rebranch merge.